src/data/directory_content.rs

```diff
@@ -15,8 +25,14 @@ use crate::config::{FileDialogConfig, FileFilter};
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct DirectoryEntry {
     path: PathBuf,
+    size: Option<u64>,
+    last_modified: Option<SystemTime>,
+    created: Option<SystemTime>,
+    file_type: Option<String>,
     is_directory: bool,
     is_system_file: bool,
+    #[cfg(feature = "metadata_view")]
+    content: Option<String>,
```

Comment:
That shouldn't be in here. Is it possible to move the content to the information panel too?

Reply:
Then it would be loaded for every single frame, which is much more resource demanding. This way it only gets loaded once with all the other metadata.
The preview images get around this by using the egui-loaders.

Reply:
I would also only load the data once in the info panel. E.g.:
```rust
if self.content.is_none() {
    // Load data
}
```

Reply:
`DirectoryContent` is kind of the same thing. This is also only loaded once in a separate thread and not every frame

Reply:
I don't really see a way to put it in InformationPanel while still keeping the ability to have users add their handler on what should happen with the content.

```rust
       // Add preview support for common text file extensions
        for text_extension in [
            "txt", "json", "md", "toml", "rtf", "xml", "rs", "py", "c", "h", "cpp", "hpp",
        ] {
            supported_files.insert(
                text_extension.to_string(),
                Box::new(|ui: &mut Ui, item: &DirectoryEntry| {
                    if let Some(mut content) = item.content() {
                        egui::ScrollArea::vertical()
                            .max_height(100.0)
                            .show(ui, |ui| {
                                ui.add(
                                    egui::TextEdit::multiline(&mut content).code_editor(),
                                );
                            });
                    }
                }) as Box<dyn FnMut(&mut Ui, &DirectoryEntry)>,
            );
        }
```

as it is used here as an attribute of the item, similar as the image loading is also done using the path attribute of the item:

```rust
       // Add preview support for JPEG and PNG image files
        supported_files.insert(
            "jpg".to_string(),
            Box::new(|ui: &mut Ui, item: &DirectoryEntry| {
                ui.label("Image");
                ui.image(format!("file://{}", item.as_path().display()));
            }) as Box<dyn FnMut(&mut Ui, &DirectoryEntry)>,
        );
```

Reply:
We could add a wrapper above `DirectoryEntry` just for the information panel. Then we can further customize the attributes. For example:
```rust
InfoPanelEntry {
    pub directory_entry: DirectoryEntry,
    pub content: Option<String>,
    ...
}
```

Reply:
Yes that could work, but I'm not sure if this is worth it?
This would move all interactions down one layer:

```rust

supported_files.insert(
            "jpg".to_string(),
            Box::new(|ui: &mut Ui, item: &InfoPanelEntry | {
                ui.label("Image");
                ui.image(format!("file://{}", item.directory_entry.as_path().display()));
            }) as Box<dyn FnMut(&mut Ui, & InfoPanelEntry)>,
        );
```

What is the motivation on why this should be moved out of `DirectoryEntry`?

Reply:
We can also implement methods like as_path in the wrapper.

I try to separate this as much as possible so that a: `DirectoryEntry` stays as small as possible and b: we can expand the information view without having to touch the internal dialog structs

Reply:
Okay, I'll try. Requires quite some rework though.
In the `InformationPanel::ui()` function we call:
```rust
// load file content and additional metadata if it's a new file
let path_option = file_dialog.active_entry();
```
and use this. I'm not sure yet, how I'll go from that to the `InfoPanelEntry`.

Reply:
Okay I tried this, can you check it?

Reply:
Thanks, I'll take a look as soon as I find time